tasks/src/main/scala/dagr/tasks/ScatterGather.scala

```diff
@@ -122,24 +173,28 @@ object ScatterGather {
     * when the partitions are available and the set of Gathers to be performed on this stage of the Scatter.
     */
   private trait SubScatter[Result <: Task] extends Scatter[Result] {
+    // the tasks that create that results
     val tasks    = new ListBuffer[Result]
+    // the sub-scatterers that further map the results
```

Comment:
Your comment implies that there _must be_ sub-scatters (scatters, not scatterers), but that's not the case.  I.e. this is the list of _any_ sub-scatters that have been created, and may well be empty.